src/Core/System/SalesChannel/Context/CartRestorer.php

```diff
@@ -93,6 +94,13 @@ private function mergeCart(Cart $customerCart, Cart $guestCart, SalesChannelCont
             return ($item->getQuantity() > 0 && $item->isStackable()) || !$customerCart->has($item->getId());
         });
 
+        $this->eventDispatcher->dispatch(new BeforeCartMergeEvent(
+            $customerCart,
+            $guestCart,
+            $mergeableLineItems,
+            $customerContext
+        ));
+
```

Comment:
Could you please add a test for these lines? And the other one that is marked in the changes tab? Otherwise I cannot merge this PR because of our 100% code coverage rule. Thank you

Reply:
What exactly should be tested? I already added one integration test here:
https://github.com/shopware/platform/pull/2759/files#diff-d5b9d48f27de3f8a86755ea4d90a93a0e067226c80f68f73de30f9e7f1b6cd45R317

Reply:
Our internal gitlab marks this as uncovered somehow, I've reimported you PR maybe it will now be marked as covered.